src/mlpack/tests/ann_layer_test.cpp

```diff
@@ -390,6 +390,133 @@ BOOST_AUTO_TEST_CASE(SimpleLinearLayerTest)
   BOOST_REQUIRE_EQUAL(arma::accu(delta), 0);
 }
 
+/**
+ * Linear module weight initialization test.
+ */
+BOOST_AUTO_TEST_CASE(LinearLayerWeightInitializationTest)
+{
+  size_t inSize = 10, outSize = 4;
+  Linear<>* linear = new Linear<>(inSize, outSize);
+  linear->Reset();
+  RandomInitialization().Initialize(linear->Weight(), outSize, inSize);
+  linear->Bias().ones();
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(linear->Weight().begin(),
+      linear->Weight().end(), linear->Parameters().begin(),
+      linear->Parameters().end() - outSize);
+
+  BOOST_CHECK_EQUAL_COLLECTIONS(linear->Bias().begin(),
+      linear->Bias().end(), linear->Parameters().begin() + inSize * outSize,
+      linear->Parameters().end());
+
+  BOOST_REQUIRE_EQUAL(linear->Weight().n_rows, outSize);
+  BOOST_REQUIRE_EQUAL(linear->Weight().n_cols, inSize);
+  BOOST_REQUIRE_EQUAL(linear->Bias().n_rows, outSize);
+  BOOST_REQUIRE_EQUAL(linear->Bias().n_cols, 1);
+  BOOST_REQUIRE_EQUAL(linear->Parameters().n_rows, inSize * outSize + outSize);
+
+  delete linear;
+}
+
+/**
+ * Atrous Convolution module weight initialization test.
+ */
+BOOST_AUTO_TEST_CASE(AtrousConvolutionLayerWeightInitializationTest)
+{
+  size_t inSize = 1, outSize = 1;
+  size_t kernelWidth = 3, kernelHeight = 3;
```

Comment:
Could you clarify why did you revert these values back? I saw there was a failing test. I'm just curious why the test had failed.

Reply:
The tests failed because I was using invalid combination of parameters, as [here](https://github.com/mlpack/mlpack/blob/22e7c8e7d24c38c6837a0486fd0a45b5e2d7900c/src/mlpack/methods/ann/layer/transposed_convolution_impl.hpp#L162-L173). 
I will look in for a correct combination.

Reply:
Thanks for the clarification.